// generator-hyfe/generators/app/templates/framework2/app/src/containers/home.js
import React, { Component, PropTypes } from 'react'
import { connect } from 'react-redux'
import { Link, browserHistory } from 'react-router'
import { loadNavs, uploadUserInfo } from '../actions/home'
import { localLogin, logout } from '../actions/user'
import Navs from '../components/nav'
import Header from '../components/header'

const styles = {
  list: {
    color: '#ccc',
    position: 'absolute',
    top: '45px',
    left: '330px',
    cursor: 'pointer'
  },
  out: {
    position: 'absolute',
    top: '45px',
    left: '260px',
    color: 'red',
    cursor: 'pointer'
  },
  fileSelect: {
    width: '700px',
    textAlign: 'center',
    paddingTop: '150px',
    background: 'azure',
    color: 'brown',
    border: '2px dashed rgb(102, 102, 102)',
    height: '200px',
    borderRadius: '5px'
  }
}

function mapStateToProps(state) {
  debugger
  return {
    items: state.home.items,
    navs: state.home.navs
  }
}

@connect(mapStateToProps, { loadNavs, logout, uploadUserInfo })
export default class Home extends Component {
  constructor(props) {
    super(props)
    this.userSignOut = this.userSignOut.bind(this)
    this.uploadFile = this.uploadFile.bind(this)
  }

  componentDidMount() {
    this.props.loadNavs()
  }

  uploadFile(acceptedFiles, rejectedFiles) {

    let form = new FormData()
    form.append('file', acceptedFiles[0])

    this.props.uploadUserInfo(form)
  }
  
  userSignOut() {
    this.props.logout()
  }

  render() {
    let navs = this.props.navs
      return (
        <div className='container'>
          <Header />
          <article className='item'>
            <Navs clickLi={this.toggleNav} navs={navs} />
            <Link style={styles.list} to='/index/list'>列表项</Link>
            <a style={styles.out} onClick={this.userSignOut}>退出</a>
            {this.props.children}
          </article>
        </div>
      )
  }
}
