// generator-hyfe/generators/app/templates/framework2/app/src/constants/actionType.js
// import keyMirror from 'keyMirror'

// export default keyMirror({
//     LOADED_ITEMS: null,
//     LOADED_ITEMS_DETAIL: null,
//     LOADED_ITEM_USER: null,
//     LOAD_CHART: null,
//     LOAD_TEST: null,
//     LOAD_NAVS: null,
//     LOGIN_SUCCESS: null
// })

//登录成功
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS'

//退出登录
export const LOGOUT_USER = 'LOGOUT_USER'

//显示错误信息
export const SHOW_ERROR_MSG = 'SHOW_ERROR_MSG'

//获取用户信息
export const GET_USERINFO = 'GET_USERINFO'
export const GET_USERINFO_SUCCESS = 'GET_USERINFO_SUCCESS'
export const GET_USERINFO_FAILURE = 'GET_USERINFO_FAILURE'

//加载最新公告列表
export const LOAD_NEW_LIST = 'LOAD_NEW_LIST'
export const LOAD_NEW_LIST_SUCCESS = 'LOAD_NEW_LIST_SUCCESS'

//加载公告详情
export const NEWS_DETAIL = 'NEWS_DETAIL'
export const NEWS_DETAIL_SUCCESS = 'NEWS_DETAIL_SUCCESS'
export const NEWS_DETAIL_FAILURE = 'NEWS_DETAIL_FAILURE'

//加载导航数据
export const LOAD_NAVS = 'LOAD_NAVS'
export const LOAD_NAVS_SUCCESS = 'LOAD_NAVS_SUCCESS'
export const LOAD_NAVS_FAILURE = 'LOAD_NAVS_FAILURE'

//上传Excel文件
export const UPLOAD_FILE = 'UPLOAD_FILE'
export const UPLOAD_FILE_SUCCESS = 'UPLOAD_FILE_SUCCESS'
export const UPLOAD_FILE_FAILURE = 'UPLOAD_FILE_FAILURE'
