// generator-hyfe/generators/app/prompts.js
var inquirer = require('inquirer')

module.exports = [
    {
        type: 'input',
        name: 'name',
        message: 'Your Project Name: ',
        default: this.appname,
        //记住用户喜好
        store: true
    }, 
    {
        type: 'checkbox',
        message: '项目目录结构: ',
        name: 'projectType',
        choices: [
            {
                value: '0',
                name: '开发框架一（基于seajs+jQuery的基础框架）',
                checked: true
            },
            {
                value: '1',
                name: '开发框架二（基于React+Redux的进阶框架）'
            }
        ]
    },
    {
        type: 'input',
        message: 'Description: ',
        name: 'description'
    },
    {
        type: 'input',
        message: 'Author: ',
        name: 'author',
        store:true
    },
    {
        type: 'input',
        message: 'Keywords: ',
        name: 'keywords'
    },
    {
        type: 'confirm',
        name: 'create',
        message: 'Are you sure you create the project now?'
    }
];
